neqo-transport/src/send_stream.rs

```diff
@@ -271,6 +271,19 @@ impl RangeTracker {
             self.used.insert(new_cur_off, (new_cur_len, cur_state));
         }
     }
+
+    /// Unmark all sent ranges.
+    pub fn unmark_sent(&mut self) {
+        let mut to_remove = SmallVec::<[_; 8]>::new();
+        for (off, (_, state)) in self.used.iter() {
+            if *state != RangeState::Acked {
+                to_remove.push(*off);
+            }
+        }
+        for r in to_remove {
+            self.used.remove(&r);
+        }
+    }
```

Comment:
We already have this guy's big brother in `unmark_range()`. Instead of this I think you can call `self.ranges.unmark_range(0, self.ranges.highest_offset())` in `TxBuffer::unmark_sent()`, perhaps also turning down the qinfos in `unmark_range` to qdebug or qtrace.